runtime/vm/stringhelpers.cpp

```diff
@@ -414,38 +414,46 @@ getStringUTF8Length(J9VMThread *vmThread, j9object_t string)
 UDATA 
 verifyQualifiedName(J9VMThread *vmThread, j9object_t string)
 {
+	J9JavaVM* vm = vmThread->javaVM;
 	UDATA unicodeLength = J9VMJAVALANGSTRING_LENGTH(vmThread, string);
 	j9object_t unicodeBytes = J9VMJAVALANGSTRING_VALUE(vmThread, string);
-	UDATA i = 0;
+	char utf8NameStackBuffer[J9VM_PACKAGE_NAME_BUFFER_LENGTH];
+	J9CfrConstantPoolInfo classNameInfo = {0, 0, 0, 0, 0, NULL, 0};
 	UDATA rc = CLASSNAME_VALID_NON_ARRARY;
+	IDATA arity = 0;
+
+	PORT_ACCESS_FROM_JAVAVM(vm);
 
-	if (unicodeLength == 0) {
+	if (0 == unicodeLength) {
 		return CLASSNAME_INVALID;
 	}
 
 	if (IS_STRING_COMPRESSED(vmThread, string)) {
 		if ('[' == J9JAVAARRAYOFBYTE_LOAD(vmThread, unicodeBytes, 0)) {
 			rc = CLASSNAME_VALID_ARRARY;
 		}
-		for (i = 0; i < unicodeLength; i++) {
-			U_8 c = J9JAVAARRAYOFBYTE_LOAD(vmThread, unicodeBytes, i);
-			if (c == '/') {
-				rc = CLASSNAME_INVALID;
-				break;
-			}
-		}
 	} else {
 		if ('[' == J9JAVAARRAYOFCHAR_LOAD(vmThread, unicodeBytes, 0)) {
 			rc = CLASSNAME_VALID_ARRARY;
 		}
-		for (i = 0; i < unicodeLength; i++) {
-			U_16 c = J9JAVAARRAYOFCHAR_LOAD(vmThread, unicodeBytes, i);
-			if (c == '/') {
-				rc = CLASSNAME_INVALID;
-				break;
-			}
-		}
 	}
+
+	classNameInfo.slot1 = (U_32)unicodeLength;
+	classNameInfo.bytes = (U_8*)copyStringToUTF8WithMemAlloc(vmThread, string,
+									J9_STR_NULL_TERMINATE_RESULT, "", 0,
```

Comment:
Have you tried passing `J9_STR_XLAT` in to `copyStringToUTF8WithMemAlloc`?  It would remove the need to add the new ` BOOLEAN isLoading)` parameter to `checkNameImpl()`

Reply:
Simply passing  `J9_STR_XLAT ` to `copyStringToUTF8WithMemAlloc `without removing the check of `isLoading` ended up with `NoClassDefFoundError `on `jdk.internal.loader.ClassLoaders$PlatformClassLoader` during the compilation as follows:
```
Exception in thread "main" java/lang/NoClassDefFoundError: jdk.internal.loader.ClassLoaders$PlatformClassLoader
        at jdk/internal/loader/ClassLoaders.<clinit> (java.base@9/ClassLoaders.java:65)
        at java/lang/ClassLoader.initializeClassLoaders (java.base@9/ClassLoader.java:176)
        at java/lang/Thread.initialize (java.base@9/Thread.java:428)
        at java/lang/Thread.<init> (java.base@9/Thread.java:153)
```
This is because `copyStringToUTF8Helper ` (called by `copyStringToUTF8WithMemAlloc `) changes `.` to `/` when `J9_STR_XLAT `is specified, in which case there should be no need to distinguish `.` and `/` in `checkNameImpl`.

That means all existing code in c`hverify.c` (including `checkNameImpl`) should remain unchanged so as to only deal with the case of `/`.  Need to check whether it works this way.

Reply:
Already updated all code involved after adding `J9_STR_XLAT `to `copyStringToUTF8WithMemAlloc `.